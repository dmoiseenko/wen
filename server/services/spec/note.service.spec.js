const noteService = require('../note.service');
const bootstrap = require('../../db/bootstrap/bootstrap.js');
const { pubsub } = require('../../graphql/pubsub.js');


describe('getNotes', () => {
  beforeEach(() => bootstrap());

  it('should get all notes', async () => {
    const actual = await noteService.getNotes();

    expect(actual).toMatchSnapshot();
  });
});

describe('createNote', () => {
  beforeEach(() => bootstrap());

  it('should return created note', async () => {
    const actual = await noteService.createNote('note text');

    expect(actual).toMatchSnapshot();
  });

  it('should publish noteAdded message with created note', (done) => {
    let subscriptionId;

    return pubsub.subscribe('noteAdded', (newNote) => {
      expect(newNote).toMatchSnapshot();
      pubsub.unsubscribe(subscriptionId);
      done();
    }).then(async (subId) => {
      subscriptionId = subId;
      await noteService.createNote('note text by subscription');
    });
  });
});

describe('getNoteById', () => {
  beforeEach(() => bootstrap());

  it('should get note by id', async () => {
    const actual = await noteService.getNoteById(1);

    expect(actual).toMatchSnapshot();
  });
});

describe('deleteNoteById', () => {
  beforeEach(() => bootstrap());

  it('should delete note by id', async () => {
    await noteService.deleteNoteById(1);
    const actual = await noteService.getNotes();

    expect(actual).toMatchSnapshot();
  });

  it('should return deleted note', async () => {
    const actual = await noteService.deleteNoteById(1);

    expect(actual).toMatchSnapshot();
  });
});
