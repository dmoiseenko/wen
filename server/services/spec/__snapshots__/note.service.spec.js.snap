// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNote should publish noteAdded message with created note 1`] = `
Object {
  "id": 4,
  "text": "note text by subscription",
}
`;

exports[`createNote should return created note 1`] = `
Object {
  "id": 4,
  "text": "note text",
}
`;

exports[`deleteNoteById should delete note by id 1`] = `
Array [
  Object {
    "id": 2,
    "text": "Look deep into nature, and then you will understand everything better. Albert Einstein",
  },
  Object {
    "id": 3,
    "text": "Be yourself; everyone else is already taken. Oscar Wilde",
  },
]
`;

exports[`deleteNoteById should return deleted note 1`] = `
Object {
  "id": 1,
  "text": "Well done is better than well said. Benjamin Franklin",
}
`;

exports[`getNoteById should get note by id 1`] = `
Object {
  "id": 1,
  "text": "Well done is better than well said. Benjamin Franklin",
}
`;

exports[`getNotes should get all notes 1`] = `
Array [
  Object {
    "id": 1,
    "text": "Well done is better than well said. Benjamin Franklin",
  },
  Object {
    "id": 2,
    "text": "Look deep into nature, and then you will understand everything better. Albert Einstein",
  },
  Object {
    "id": 3,
    "text": "Be yourself; everyone else is already taken. Oscar Wilde",
  },
]
`;
